omnidoc/lark_tree_matcher.py

```diff
@@ -0,0 +1,154 @@
+"""Tree matcher based on Lark grammar
+
+TODO: Move to lark-parser, refactor reconstruct.py to use this as base class.
+"""
+
+
+from collections import defaultdict
+
+from lark.tree import Tree
+from lark.common import ParserConf
+from lark.lexer import Token
+from lark.parsers import earley
+from lark.grammar import Rule, Terminal, NonTerminal
+
+
+def is_discarded_terminal(t):
+    return t.is_term and t.filter_out
+
+def is_iter_empty(i):
+    try:
+        _ = next(i)
+        return False
+    except StopIteration:
+        return True
+
+
+class MakeTreeMatch:
```

Comment:
Please add a doc string on what this supposed to do? One line summary and explanation of init parameters would be nice.

Reply:
Recommend this extension: https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring